### ISSUE TITLE
[IOS-XE]parser for 'show interface {interface name} ' flow_control issue

### ISSUE THREAD

**Issue by hsang88** [CiscoTestAutomation/genieparser#860]
in a IOS-XE device, cli output

show interfaces g0/0/0

> GigabitEthernet0/0/0 is up, line protocol is up 
>   Hardware is C1111-2x1GE, address is b08b.cf58.ee80 (bia b08b.cf58.ee80)
>   MTU 1500 bytes, BW 1000000 Kbit/sec, DLY 10 usec, 
>      reliability 255/255, txload 1/255, rxload 1/255
>   Encapsulation ARPA, loopback not set
>   Keepalive not supported 
>   Full Duplex, 1000Mbps, link type is force-up, media type is RJ45
>   **output flow-control is off, input flow-control is on**
>   ARP type: ARPA, ARP Timeout 04:00:00
>   Last input 02:25:34, output 00:00:08, output hang never
>   Last clearing of "show interface" counters 4d01h
>   Input queue: 0/375/0/0 (size/max/drops/flushes); Total output drops: 0
>   Queueing strategy: fifo
>   Output queue: 0/40 (size/max)
>   5 minute input rate 0 bits/sec, 0 packets/sec
>   5 minute output rate 0 bits/sec, 0 packets/sec
>      263 packets input, 15780 bytes, 0 no buffer
>      Received 263 broadcasts (0 IP multicasts)
>      0 runts, 0 giants, 0 throttles 
>      0 input errors, 0 CRC, 0 frame, 0 overrun, 0 ignored
>      0 watchdog, 0 multicast, 0 pause input
>      5821 packets output, 2080975 bytes, 0 underruns
>      Output 101 broadcasts (0 IP multicasts)
>      0 output errors, 0 collisions, 69 interface resets
>      0 unknown protocol drops
>      0 babbles, 0 late collision, 0 deferred
>      95 lost carrier, 0 no carrier, 0 pause output
>      0 output buffer failures, 0 output buffers swapped out

use parser to get flow control result

parsed = dev.parse('show int g0/0/0')

> 
> {'GigabitEthernet0/0/0': {'arp_timeout': '04:00:00',
>                           'arp_type': 'arpa',
>                           'auto_negotiate': False,
>                           'bandwidth': 1000000,
>                           'counters': {'in_broadcast_pkts': 263,
>                                        'in_crc_errors': 0,
>                                        'in_errors': 0,
>                                        'in_frame': 0,
>                                        'in_giants': 0,
>                                        'in_ignored': 0,
>                                        'in_mac_pause_frames': 0,
>                                        'in_multicast_pkts': 0,
>                                        'in_no_buffer': 0,
>                                        'in_octets': 15780,
>                                        'in_overrun': 0,
>                                        'in_pkts': 263,
>                                        'in_runts': 0,
>                                        'in_throttles': 0,
>                                        'in_watchdog': 0,
>                                        'last_clear': '4d02h',
>                                        'out_babble': 0,
>                                        'out_broadcast_pkts': 101,
>                                        'out_buffer_failure': 0,
>                                        'out_buffers_swapped': 0,
>                                        'out_collision': 0,
>                                        'out_deferred': 0,
>                                        'out_errors': 0,
>                                        'out_interface_resets': 69,
>                                        'out_late_collision': 0,
>                                        'out_lost_carrier': 95,
>                                        'out_mac_pause_frames': 0,
>                                        'out_multicast_pkts': 0,
>                                        'out_no_carrier': 0,
>                                        'out_octets': 2149313,
>                                        'out_pkts': 6003,
>                                        'out_underruns': 0,
>                                        'out_unknown_protocl_drops': 0,
>                                        'rate': {'in_rate': 0,
>                                                 'in_rate_pkts': 0,
>                                                 'load_interval': 300,
>                                                 'out_rate': 0,
>                                                 'out_rate_pkts': 0}},
>                           'delay': 10,
>                           'duplex_mode': 'full',
>                           'enabled': True,
>                           'encapsulations': {'encapsulation': 'arpa'},
>                           **'flow_control': {'receive': False, 'send': True}**,
>                           'is_deleted': False,
>                           'last_input': '03:24:18',
>                           'last_output': '00:00:11',
>                           'line_protocol': 'up',
>                           'link_type': 'force-up',
>                           'mac_address': 'b08b.cf58.ee80',
>                           'media_type': 'RJ45',
>                           'mtu': 1500,
>                           'oper_status': 'up',
>                           'output_hang': 'never',
>                           'phys_address': 'b08b.cf58.ee80',
>                           'port_channel': {'port_channel_member': False},
>                           'port_speed': '1000mbps',
>                           'queues': {'input_queue_drops': 0,
>                                      'input_queue_flushes': 0,
>                                      'input_queue_max': 375,
>                                      'input_queue_size': 0,
>                                      'output_queue_max': 40,
>                                      'output_queue_size': 0,
>                                      'queue_strategy': 'fifo',
>                                      'total_output_drop': 0},
>                           'reliability': '255/255',
>                           'rxload': '1/255',
>                           'txload': '1/255',
>                           'type': 'C1111-2x1GE'}}

get flow_control result only:
parsed['GigabitEthernet0/0/0']['flow_control']

> {'receive': False, 'send': True}

compare to original cli output 
output flow-control is off, input flow-control is on
I think current show interface parse get opposite key 
the 'receive' should align with input
the 'send' should align with output

please check it. thanks.

**Comment by Harishv01** [comment]
Hi,

Kindly give me some time. I am checking it now and will get back as soon as possible.

**Comment by Harishv01** [comment]
I have raised a PR for this issue and will let you know once it is merged. When you raise a ticket, please make sure to add the command properly.

**Comment by hsang88** [comment]
get it, thanks.

**Comment by Harishv01** [comment]
Hi,

I have merged this PR and the fix will be available in version 24.6. Hence, I am closing the issue.

Thank you.

**Comment by hsang88** [comment]
thanks, I will verify it in version 24.6

**Comment by hsang88** [comment]
verify ok with version 24.6
